internal/addr.go

```diff
@@ -26,16 +26,12 @@ import (
 const defaultHost = "127.0.0.1"
 
 func ParseAddr(addr string) (string, int, error) {
-	// TODO: refactor this function, it can be written in a better way.
 	if addr == "" || strings.TrimSpace(addr) == "" {
 		return defaultHost, 0, nil
 	}
-	if !strings.Contains(addr, ":") {
-		return addr, 0, nil
-	}
 	host, port, err := net.SplitHostPort(addr)
-	if err != nil {
-		return "", 0, err
+	if err != nil { // return the address itself if port is missing in the address
+		return addr, 0, nil
```

Comment:
Returning the address on an error doesn't look correct to me. It would be better to detect whether the address is ipv6 without a port and return early but still return an error if there's a problem with parsing the address.

Reply:
I agree with you. It is not the best practice as It ignores the errors. It will lead to more potential bug because It doesn't cover any undesired cases.